extension.js

```diff
@@ -141,8 +170,57 @@ class PipOnTop
       /* Change stick if enabled or unstick PipAble windows */
       un = (isPipWin && this.settings.get_boolean('stick')) ? '' : 'un';
       window[`${un}stick`]();
+
+      /* Repeatedly override new window position so it sticks */
+      window._overrideTimeoutId = GLib.timeout_add(GLib.PRIORITY_DEFAULT, 1000, () => {
+        window._overrideTimeoutId = null;
+        return false;
+      });
     }
   }
+
+  _onWindowChanged(window)
+  {
+    if (!window._isPipAble)
+      return;
+
+    /* Override new window position and size until timeout */
+    if (window._overrideTimeoutId) {
+      if (this._lastWindowRect) {
+        let last = this._lastWindowRect;
+        let current = window.get_frame_rect();
+        /* Change position independently of size to avoid aspect
+         * ratio lock interference */
+        window.move_resize_frame(false, last.x, last.y,
+                                 current.width, current.height);
+        /* Only care about height but width also needs to be applied
+         * to avoid window shrinking (Firefox Bug 1794577) */
+        window.move_resize_frame(false, last.x, last.y,
```

Comment:
AFAIU, you are changing window when window gets changed and this whole part of code runs constantly thousands of times within 1 second until timeout guarding it is gone... Won't this create short (1 sec) UI freezes and ramp up CPU usage?